<!DOCTYPE html>
<html lang="en">
  <head>
    <title>ProDy &mdash; Protein Dynamics and Sequence Analysis</title>

    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <link href="{{ pathto('_static/css/bootswatch-united.min.css', 1) }}" rel="stylesheet">
    <link href="{{ pathto('_static/css/prodydocs.css', 1) }}" rel="stylesheet">

    <script type="text/javascript" src="{{ pathto('_static/js/jquery-2.0.0.min.js', 1) }}"></script>
    <!--<script type="text/javascript" src="{{ pathto('_static/js/bootstrap.min.js', 1) }}"></script> -->

  <!-- Latest compiled and minified JavaScript -->
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js" integrity="sha384-0mSbJDEHialfmuBBQP6A4Qrprq5OVfW37PRR3j5ELqxss1yVqOtnepnHVP9aJ7xS" crossorigin="anonymous"></script>
    <script type="text/javascript" src="{{ pathto('_static/js/prodydocs.js', 1) }}"></script>
    <script type="text/javascript">

      var _gaq = _gaq || [];
      _gaq.push(['_setAccount', 'UA-19801227-1']);
      _gaq.push(['_trackPageview']);

      (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
      })();

    </script>

<!-- Statistics scripts for webpage --> 

    <script type="text/javascript" src="https://www.google.com/jsapi"></script>
    <script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
    <script type="text/javascript">
    google.charts.load('current', {'packages':['geochart','table','corechart']});
    google.charts.setOnLoadCallback(drawRegionsMap);
    function drawRegionsMap() {
        var quere = new google.visualization.Query('https://cors-anywhere.herokuapp.com/https://prody-users.appspot.com/query?id=ag1zfnByb2R5LXVzZXJzchULEghBcGlRdWVyeRiAgICA3pCBCgw&format=data-table-response'
        );

        quere.send(function (response) {
           var datae = response.getDataTable();

        datae.setColumnLabel(0, 'Country');
        datae.setColumnLabel(1, 'Sessions');

        var options = {};
        options['dataMode'] = 'regions';

        var container = document.getElementById('regions_div');
        var geomap = new google.visualization.GeoChart(container);

        geomap.draw(datae, options);
        });
   };
    google.charts.setOnLoadCallback(drawTable);
    function drawTable() {
        var querf = new google.visualization.Query('https://cors-anywhere.herokuapp.com/https://prody-users.appspot.com/query?id=ag1zfnByb2R5LXVzZXJzchULEghBcGlRdWVyeRiAgICA2reXCgw&format=data-table-response'
        );

        querf.send(function (response) {
           var dataf = response.getDataTable();

        dataf.setColumnLabel(0, 'Country');
        dataf.setColumnLabel(1, 'Sessions');

        var table = new google.visualization.Table(document.getElementById('table_div'));

        table.draw(dataf, {showRowNumber: true, width: '100%', height: '100%'});
        });
    };
    google.charts.setOnLoadCallback(drawTable2);
    function drawTable2() {
        var querd = new google.visualization.Query('https://cors-anywhere.herokuapp.com/https://prody-users.appspot.com/query?id=ag1zfnByb2R5LXVzZXJzchULEghBcGlRdWVyeRiAgICAgICACgw&format=data-table-response'
        );
        querd.send(function (response) {
           var datad = response.getDataTable();

        datad.setColumnLabel(0, 'Unique Visitors');
        datad.setColumnLabel(1, 'Visits');
        datad.setColumnLabel(2, 'Unique Visitors/Visits');
        datad.setColumnLabel(3, 'Avg. Pages/Visit');
        datad.setColumnLabel(4, 'Avg. Duration/Visit');

        for (var c=0, length = datad.getNumberOfRows(), myVal; c < length; c++) {
            myVal = parseFloat(datad.getValue(c, 1)).toFixed(1);
            datad.setFormattedValue(c, 2, myVal + '%');
          }
for (var c=0, length = datad.getNumberOfRows(), myVal; c < length; c++) {
            myVal = parseFloat(datad.getValue(c, 2)).toLocaleString('en');
            datad.setFormattedValue(c, 1, '       ' + myVal );
          }
for (var i = 0, length = datad.getNumberOfRows(), value, minutes, 
 seconds; i < length; i++) {
     value = datad.getValue(i, 3);
     minutes = Math.floor(value / 60);
     seconds = Math.round(value % 60);
     // prepend 0's if necessary
     minutes = (minutes < 10) ? '0' + minutes : minutes;
     seconds = (seconds < 10) ? '0' + seconds : seconds;
     datad.setFormattedValue(i, 4,      minutes + ':' + seconds + ' (min:sec)');
}
for (var c=0, length = datad.getNumberOfRows(), myVal; c < length; c++) {
            myVal = parseFloat(datad.getValue(c, 4)).toFixed(1);
            datad.setFormattedValue(c, 3, myVal + ' (number of pages)');
          }

        var table2 = new google.visualization.Table(document.getElementById('table_div2'));

        table2.draw(datad, { width: '100%', height: '100%' });
        });
    };

    google.charts.setOnLoadCallback(drawChart)
    function drawChart() {
        var querf = new google.visualization.Query('https://cors-anywhere.herokuapp.com/https://prody-users.appspot.com/query?id=ag1zfnByb2R5LXVzZXJzchULEghBcGlRdWVyeRiAgICAmdKFCgw&format=data-table-response'
        );

        querf.send(function (response) {
           var dataf = response.getDataTable();

        dataf.setColumnLabel(0, 'Country');
        dataf.setColumnLabel(1, 'Sessions');

        var options = {
          title: 'Visitor distribution across the world (top 20 countries)',
          titleTextStyle: {
                fontSize: 12
        },
          is3D: true,
          height: '264', 
          chartArea: {'width':'100%', 'height':'80%'}
        };

        var chart = new google.visualization.PieChart(document.getElementById('piechart1'));

        chart.draw(dataf, options);
        });
      };
    google.charts.setOnLoadCallback(drawChart2)
    function drawChart2() {

        var querg = new google.visualization.Query('https://cors-anywhere.herokuapp.com/https://prody-users.appspot.com/query?id=ag1zfnByb2R5LXVzZXJzchULEghBcGlRdWVyeRiAgICA7bGDCgw&format=data-table-response'
        );

        querg.send(function (response) {
           var datag = response.getDataTable();

        datag.setColumnLabel(0, 'Country');
        datag.setColumnLabel(1, 'Unique Visitors');

         var options = {
          title: 'Unique visitors across the world (top 20 countries)',
          titleTextStyle: {
                fontSize: 12
        },
          is3D: true,
          height: '264', 
          chartArea: {'width':'100%', 'height':'80%'}
        };

        var chartg = new google.visualization.PieChart(document.getElementById('piechart2'));

        chartg.draw(datag, options);
        });
      };
    </script>

    <link rel="shortcut icon" href="{{ pathto('_static/favicon.ico', 1) }}"/>

    <style type="text/css">
      .slide-title {
        padding-left: 10px;
        width: 290px;
        font-weight: bold;
        font-size: 90%;
        text-align: left;
        float: left;
      }

      .slide-arrow {
        cursor: pointer;
        float: right;
        /*position: absolute;*/
        top: 0px;
        vertical-align: top;
      }

      .slide-content {
        padding-top: 30px;
      }

      .slide-content > img {
        margin-left: auto;
        margin-right: auto;
      }

      .slide-image {
        text-align: center;
      }

      .prody-slide {
        display: none;
        visibility: hidden;
      }

      body {
        margin-left: 40px;
        margin-right: 40px;
      }

    </style>

    <!-- Figures
    ================================================== -->
    
    <script type="text/javascript">
      slideTitles = new Array(
      "Compare Dynamics from Experiments and Theory",
      "<i>Evol</i> for Sequence Evolution Analysis",
      "Evolution and Dynamics of Uracil-DNA Glycosylase",
      "Evolution and Dynamics of Hsp70 ATPase domain",
      "Dynamics of p38 MAP Kinase",
      "Compare Dynamics from Experiments and Theory",
      "Compare Experiments, Simulation, and Theory",
      "Intrinsic Dynamics of HIV-1 Reverse Transcriptase",
      "Quick Representations in Interactive Sessions",
      "Depicting Normal Modes Couldn't Be Easier",
      "Depicting Normal Modes Couldn't Be Easier",
      "Compare Dynamics from Experiments and Theory",
      "Druggability Simulation of Kinesin Eg5",
      "Kinesin Eg5 Druggable Binding Sites",
      "Leucine Transporter (LeuT) Energy Landscape", 
      "Collective Molecular Dynamics of Adenylate Kinase",
      "Visualization of Collective Molecular Dynamics",
      "OF and IF structures of Glt<sub>Ph</sub>",
      "Motion of OF structure of Glt<sub>Ph</sub>",
      "Motion of IF structure of Glt<sub>Ph</sub>",
      "Deformability Profile of Ubiquitin",
      "Mean Value of Effective Spring Constant",
      "Mechanical Stiffness Map with Effective Force Constant",
      "Applying Gaussian Network Model on Hi-C data",
      "Normal Mode Analysis of Human Chromosomes",
      "Spectral Clustering Aids in Identification of Structural Domains",
      "Interdomain allostery in Hsp70",
      "Sensors and effectors in AMPARs",
      "Interdomain signaling in AMPARs",
      "Signature global modes and cross-correlations for the LeuT fold family",
      "Comparing signature square fluctuations across LeuT family members",
      "Projection of PBP-1 family members onto the subspace of 2 global modes",
      "Frequency distribution of global modes among family members",
      "Pharmmaker pipeline showing steps for virtual screening from druggability simulations",
      "Pharmmaker uses snapshots from druggability simulations to guide pharmacophore model making",
      "High affinity residue analysis example for an AMPAR LBD dimer simulation",
      "Essential sites of glutamate racemase",
      "Identification of alternative ligand-binding sites on GPCRs",
      "Prediction of allosteric pocket in beta-lactamase",
      "TRiC/CCT Cryo-EM map fitted with TRN pseudoatom beads",
      "Global dynamics of CCT-ATP from TRN ENM",
      "ANM mode 7 of TRiC-ATP",
      "Schematic description of symmetric or anti-symmetric movements of CCT-ATP",
      "Population distribution of ClustENMD conformers of adenylate kinase (AK)",
      "Conformational surface of HIV-1 reverse transcriptase",
      "Conformational surface of HIV-1 reverse transcriptase by generation",
      "Detecting non-covalent interactions in  USP5 zinc-finger ubiquitin binding domain",
      "Ensemble analysis of interactions",
      "Residues subject to high numbers of interactions",
      "Detecting pairs of residues linked by water bridges in LMW-PTP protein",
      "Determining prevalent water clusters in the PE-binding protein 1",
      "Frequency of interactions with water molecules for a LMW-PTP protein trajectory",
      "WatFinder prediction of water clusters in membrane transporter",
      "WatFinder prediction of water influx in membrane transporter",
      "Scipion-EM-ProDy Ensemble Analysis",
      "Scipion-EM-ProDy GNM Analysis",
      "ANMD sampling of mGluR1 NTD along mode 1",
      "ANMD sampling of mGluR1 NTD along mode 2",
      "RBD-VH F6 Motion from ANMD"
      );

      panelSlides = new Array();
      panelSlides[0] = new Array(0, 4, 1, 7);
      panelSlides[1] = new Array(4, 5, 6, 8);
      panelSlides[2] = new Array(1, 2, 3);
      panelSlides[3] = new Array(7, 9, 10, 11);
      panelSlides[4] = new Array(17, 18, 19);
      panelSlides[5] = new Array(20, 21, 22);
      panelSlides[6] = new Array(12, 13);
      panelSlides[7] = new Array(14, 15, 16);
      panelSlides[8] = new Array(23, 24, 25);
      panelSlides[9] = new Array(26, 27, 28);
      panelSlides[10] = new Array(29, 30, 31, 32);
      panelSlides[11] = new Array(33, 34, 35);
      panelSlides[12] = new Array(36, 37, 38);
      panelSlides[13] = new Array(39, 40,  42);
      panelSlides[14] = new Array(43, 44, 45);
      panelSlides[15] = new Array(46, 47, 48);
      panelSlides[16] = new Array(49, 50, 51, 52, 53);
      panelSlides[17] = new Array(54, 55);
      panelSlides[18] = new Array(56, 57, 58);

      var slideIndices = [0, 0, 0, 0, 0, 0, 0, 0, 0];

      function nextSlide(panel) {
        showSlide(panel, (slideIndices[panel] + 1) % panelSlides[panel].length);
      }

      function prevSlide(panel) {
        len = panelSlides[panel].length;
        showSlide(panel, (slideIndices[panel] - 1 + len) % len);
      }

      function showSlide(panel, which) {

        if (which < 0) {
          which = Math.floor(Math.random() * (panelSlides[panel].length));
        }
        slideIndices[panel] = which;
        index = panelSlides[panel][which];
        document.getElementById("panel-" + panel + "-title").innerHTML =
        slideTitles[index] + " (" + (which + 1) + "/" + panelSlides[panel].length + ")";
        document.getElementById("panel-" + panel + "-slide").innerHTML =
        document.getElementById("slide-" + index).innerHTML;
      }


      $(document).ready(function() {
        var down = $("#download");
        if (navigator.appVersion.indexOf("Win") >= 0) {
         var sel = $("#downloads > li > a")[2];
        } else {
          var sel = $("#downloads > li > a")[0];
        }
        var sel = $(sel);
        down.attr("href", sel.attr("href"))
        down.html('<i class="icon-download-alt icon-white"></i> ' + sel.html())

      });



    </script>

  </head>

  <!-- Google tag (gtag.js) -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-LE3T4FZ61C"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'G-LE3T4FZ61C');
  </script>

  <body id="top" data-offset="190">

      <div class="prody-slide" id="slide-0">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/ProDy_Bioinf_Fig1.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Dynamics of p38 MAP kinase inferred from a structural ensemble
        using PCA is compared to intrinsic dynamics
        of the protein modeled using ANM.
        See
        <a href="{{ pathto('tutorials/ensemble_analysis/xray') }}">PCA of X-ray structures</a>
        or
        <a target="_blank" href="http://bioinformatics.oxfordjournals.org/content/27/11/1575.full">Bioinformatics article</a>
        for more details.
        </p>
      </div>

      <div class="prody-slide" id="slide-1">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/Evol_MBE_Fig1.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Workflow for comparative analysis of sequence evolution and structural dynamics is shown.
        See <a href="manual/apps/evol/index') }}">Evol Applications</a>
        or
        <a target="_blank" href="http://mbe.oxfordjournals.org/content/29/9/2253.long">Mol Biol Evol article</a>
        for more details.
        </p>
      </div>

      <div class="prody-slide" id="slide-2">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/Evol_MBE_Fig2.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Results from comparative analysis of residue conservation,
        conformational mobility, and coevolutionary patterns for uracil-DNA glycosylase.
        See
        <a target="_blank" href="http://mbe.oxfordjournals.org/content/29/9/2253.long">Mol Biol Evol article</a>
        or
        <a href="{{ pathto('tutorials/evol_tutorial/msaanalysis') }}">Conservation and Coevolution Analysis</a>
        for more details.
        </p>
      </div>

      <div class="prody-slide" id="slide-3">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/Evol_PLoS_Fig6.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Coevolution of NEF-binding residues analzyed using mutual information is displayed for the Hsp70 ATPase domain.
        See
        <a target="_blank" href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1000931">PLoS Comp Biol article</a>
        or
        <a href="{{ pathto('tutorials/evol_tutorial/msaanalysis') }}">Conservation and Coevolution Analysis</a>
        for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-4">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/ProDy_PNAS_Fig2.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Comparative analysis of p38 MAP kinase dynamics from experiments (PCA) and theory (ANM).
        See the PNAS
        <a target="_blank" href="http://www.pnas.org/content/106/34/14349.full">article</a>
        or
        <a target="_blank" href="http://www.pnas.org/content/106/34/14349/F2.expansion.html">figure</a>
        for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-5">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/ProDy_ProtSci_Fig3.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Comparative analysis of dynamics of drug target proteins and model systems from experiments (PCA) and theory (ANM).
        See the Protein Science
        <a target="_blank" href="http://onlinelibrary.wiley.com/doi/10.1002/pro.711/full">article</a>
        for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-6">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/ProDy_ProtSci_Fig4.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Comparative analysis of p38 MAP kinase dynamics from experiments (PCA), simulations (EDA), and theory (ANM).
        See the Protein Science
        <a target="_blank" href="http://onlinelibrary.wiley.com/doi/10.1002/pro.711/full">article</a>
        for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-7">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/nmwiz_youtube.gif', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Animation shows HIV-1 reverse transcriptase functional motions calculated
        using anisotropic network model.
        Arrows and animations are generated using
        <a href="{{ pathto('tutorials/nmwiz_tutorial/intro') }}">NMWiz</a>
        VMD plugin.  See
        <a href="{{ pathto('tutorials/nmwiz_tutorial/index') }}">NMWiz tutorial</a>
        for usage examples.
        </p>
      </div>

      <div class="prody-slide" id="slide-8">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/ProDy_protein.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        You can make a quick protein representation in interactive sessions using
      <a href="{{ pathto('manual/reference/proteins/functions.html#prody.proteins.functions.showProtein', 1) }}">showProtein()</a>
        function.
        </p>
      </div>

      <div class="prody-slide" id="slide-9">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/NMWiz.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        NMWiz is designed for picturing normal modes easy. Image shows arrows from slowest three ANM modes for p38 MAP kinase centered at the origin.
        They indeed align with planes normal to each other.
        </p>
      </div>

      <div class="prody-slide" id="slide-10">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/NMWiz_net3m.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        NMWiz makes depicting elastic network models and protein motions predicted with
        them easy. Image shows ANM model for p38 MAP kinase and three slow ANM modes (below).
        </p>
      </div>

      <div class="prody-slide" id="slide-11">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/NMWiz_compare.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        NMWiz can be used to comparative dynamics inferred from experimental datasets and predicted using theory.
        </p>
      </div>

      <div class="prody-slide" id="slide-12">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/drugui_youtube.gif', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        The movie shows a molecular dynamics simulation
        for assessing the druggability of kinesin eg5.
        <a href="{{ pathto('tutorials/nmwiz_tutorial/intro') }}">NMWiz</a>
        VMD plugin.  See
        <a href="{{ pathto('tutorials/nmwiz_tutorial/index') }}">NMWiz tutorial</a>
        for usage examples.
        </p>
      </div>

      <div class="prody-slide" id="slide-13">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/eg5_druggability.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Kinesin Eg5 druggable sites, including allosteric inhibitor binding site and
        and tubulin binding site, identified by simulations are shown. See our
        <a href="http://pubs.acs.org/doi/full/10.1021/ct300117j">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-14">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/comd_ake-new.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Sampling of the functional substates (inward-facing (IF) or outward-facing (OF),  in closed (c) or open (o) forms) of LeuT using coMD simulations. See <a href="http://scitation.aip.org/content/aip/journal/jcp/143/24/10.1063/1.4936133">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-15">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/comd_youtube2.gif', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        The movie illustrates a coMD trajectory for adenylate kinase. <a href="{{ pathto('tutorials/nmwiz_tutorial/intro.html', 1) }}">NMWiz</a> VMD plugin. See <a href="{{ pathto('tutorials/nmwiz_tutorial/index.html', 1) }}">NMWiz tutorial</a> for usage examples.
        </p>
      </div>

      <div class="prody-slide" id="slide-16">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/comd_pca_path_slide.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Energy landscape in the space of principal coordinates.
        </p>
      </div>

      <div class="prody-slide" id="slide-17">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/memanm-ofif.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Outward-facing (OF) and inward-facing (IF) structures of Glt<sub>Ph</sub> show a large displacement of the core domains. 
        See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3309413/">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-18">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/OF_imANM_mode2.gif', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        The second mode of the OF structure moves all three transport domains simultaneously through the membrane in a ‘lift-like’ motion.
        See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3309413/">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-19">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/IF_imANM_mode2.gif', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        The second mode of the IF structure moves all three transport domains simultaneously through the membrane in a ‘lift-like’ motion.
         See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3309413/">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-20">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/1ubi_meanSM.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Deformability profile of ubiquitin (PDB code: 1UBI). Structure is automatically uploaded to VMD program where blue color shows regions which are mechanically more resistant to the external force.  
        </p>
      </div>

      <div class="prody-slide" id="slide-21">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/meanPlot_1ubi.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Mean value of effective spring constant (calculated from mechanical stiffness matrix) with secondary structure of ubiquitin. Blue color indicates mechanically strong regions. 
        </p>
      </div>

      <div class="prody-slide" id="slide-22">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/1ubi_map.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Mechanical Stiffness Map with effective force constant in a color bar (blue - strong regions, red - weak regions) for ubiquitin.
        </p>
      </div>

      <div class="prody-slide" id="slide-23">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/hic_workflow.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Workflow for GNM analysis of chromatin dynamics. 
        See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5397156/">publication</a> for details. 
        </p>
      </div>

      <div class="prody-slide" id="slide-24">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/hic_domains.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Covariance matrix of chromosome 17 of human B cells. Structural domains and CCDDs are identified and outlined. 
        See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5397156/">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-25">
        <div class="slide-image"><img src="{{ pathto('_static/gallery/embed_chromo.gif', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          3D Laplacian embedding of chromosome 17 loci using the first three principal modes. 
          See ChromD tutorial for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-26">
        <div class="slide-image"><img src="{{ pathto('_static/figures/PRS_General_2014_pcbi.figure3.jpg', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Perturbation response scanning of the Hsp70 chaperone reveals interdomain allostery. 
        See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4022485/">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-27">
        <div class="slide-image"><img src="{{ pathto('_static/figures/PRS_Dutta_2015_Structure.figure6_highres.jpg', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Perturbation response scanning of the AMPA-type glutamate receptor reveals sensors and effectors for allosteric signaling. 
        See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4558295/">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-28">
        <div class="slide-image"><img src="{{ pathto('_static/figures/PRS_Dutta_2015_Structure.figure7_highres.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        A more in-depth analysis of the PRS matrix reveals interdomain signaling in the AMPA receptor.
        See <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4558295/">publication</a> for details.
        </p>
      </div>

      <div class="prody-slide" id="slide-29">
        <div class="slide-image"><img src="{{ pathto('_static/figures/SignDy_LeuT_main_fig.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          The left panels show the three softest GNM modes (blue lines) and their standard deviations (faint blue bands). 
          Red and blue regions in the corresponding ribbon diagrams show regions moving in opposite directions. 
          The right panel has the average cross-correlation matrix from the first 20 global modes (top) and its standard deviation (bottom).
        </p>
      </div>

      <div class="prody-slide" id="slide-30">
        <div class="slide-image"><img src="{{ pathto('_static/figures/SignDy_LeuT_offset_profiles.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
        Square fluctuations calculated from the top 5 global modes are shown for a number of LeuT fold family 
        members, revealing similarities and subfamily- or conformation-dependent differences. 
        </p>
      </div>

      <div class="prody-slide" id="slide-31">
        <div class="slide-image"><img src="{{ pathto('_static/figures/SignDy_PBP-1_anm_projection_2_modes_annotated.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Type-I periplasmic binding protein domains are mapped onto the first two signature ANM modes. 
          These domains, found in a range of proteins including bacterial solute carriers and eukaryotic receptors, 
          have two lobes that undergo well-characterised conserved motions that are evident from comparison of structures. 
          SignDy reveals such conserved dynamics.
        </p>
      </div>

      <div class="prody-slide" id="slide-32">
        <div class="slide-image"><img src="{{ pathto('_static/figures/SignDy_TIM_variance_distributions.png', 1) }}"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          SignDy analysis allows a comparison of the frequency dispersion of family members. 
          The distribution of inverse eigenvalues is shown for the softest five modes for TIM barrel fold family.
        </p>
      </div>

      <div class="prody-slide" id="slide-33">
        <div class="slide-image"><img src="../_static/figures/pharmmaker-figure1.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Pharmmaker (center) includes four main programs (steps 2 to 5 of the pipeline)
          that bridge druggability simulations from DruGUI (left) 
          with pharmacophore-based virtual screening (right).
          The names of the programs are given under each step in blue.
        </p>
      </div>

      <div class="prody-slide" id="slide-34">
        <div class="slide-image"><img src="../_static/figures/pharmmaker.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          (Left) A snapshot extracted from druggability simulations for an AMPAR LBD dimer
          using Pharmmaker including probe poses and target conformation.
          Dominant binding interactions between probe and residues are shown.
          (Right) A pharmacophore model built based on the snapshot.
          One hydrogen acceptor, one donor, and two hydrophobic features were used to represent the probes.
        </p>
      </div>

      <div class="prody-slide" id="slide-35">
        <div class="slide-image"><img src="../_static/figures/pharmmaker-figure3A-B.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Each residue is given a binding value for each probe type, based on an inverse square distance potential.
          This is shown in the two graphs for the two subunits of an AMPAR LBD dimer.
          The dotted lines indicate a cutoff of 500, above which residues are defined as high affinity residues 
          for a particular probe.
        </p>
      </div>

      <div class="prody-slide" id="slide-36">
        <div class="slide-image"><img src="../_static/figures/essa_1.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          ESSA profile (A)  gives a measure of the extent of frequency shift in the global modes induced by each residue. 
          Residues (red circles) interacting with the allosteric ligand (PDB id: 2jfn) correspond to essential sites. 
          Two differentperspectives (B-C) display color-coded by z-scores from red (highest) to blue (lowest) together with bound ligands.
        </p>
      </div>

      <div class="prody-slide" id="slide-37">
        <div class="slide-image"><img src="../_static/figures/essa_2.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          ESSA results for (A,D) muscarinic acetylcholine receptor and (B,C) free fatty acid receptor 1 GPR40. 
          GPCRs are color-coded by the ESSA profile. Various allosteric ligand binding sites, 
          as well as the G-protein (pink) binding site, correspond to essential or hot regions.
        </p>
      </div>

      <div class="prody-slide" id="slide-38">
        <div class="slide-image"><img src="../_static/figures/essa_3.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Complex structure (PDB id: 1pzo) resolved in the presence of two allosteric ligands 
          (spatially neighboring, both shown in magenta sticks) and the orthosteric ligand
          (yellow sticks). The meshed surface displays the predicted allosteric pocket enclosing all ligands.
        </p>
      </div>

      <div class="prody-slide" id="slide-39">
        <div class="slide-image"><img src="../_static/figures/8000nodes_with_map.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Superposition of the TRiC-AMP-PNP electron density map (EMD-1961; grey surface) and 8000 TRN nodes fitted to it (pink, spheres). 
          This number of nodes corresponds to 1 residue/node, but higher levels of coarse-graining such as 3000 nodes works too.
        </p>
      </div>

      <div class="prody-slide" id="slide-40">
        <div class="slide-image"><img src="../_static/figures/cryody_fig_3.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          (A) Results from ANM analysis of the TRN (based on EMD-1961), displaying the architecture colour-coded by the MSFs of nodes (blue: most rigid; orange: most mobile) in the softest 20 modes. 
          (B) MSFs of the subunits as driven by the subsets of 5 (green), 10 (orange) and 20 (blue) softest modes. (C) Covariance between the global motions of the subunits based on the softest 20 modes. 
          (D) Orientational correlations between the global movements of the subunits.
        </p>
      </div>

      <div class="prody-slide" id="slide-41">
        <div class="slide-image"><img src="../_static/figures/cryody_fig_5.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          The beads reconstructed from EMD-1961 are displaced along ANM mode 7 with an RMSD of 6 Å in both directions, revealing motions related to upper ring closure. 
        </p>
      </div>

      <div class="prody-slide" id="slide-42">
        <div class="slide-image"><img src="../_static/figures/cryody_fig_5.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          (A-C) Symmetric and anti-symmetric movements for circularly symmetric shapes. Gray and orange arrows indicate alternating motions between symmetric expansion and compression in A, 
          stretching and contraction along orthogonal directions in B, and opposite direction rotations in C. (D-F) Modes 6, 1 and 7 of CCT/TRiC (upper ring) approximate the above motions, respectively.
        </p>
      </div>

      <div class="prody-slide" id="slide-43">
        <div class="slide-image"><img src="../_static/figures/clustenmd_fig_1.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Population distribution of ClustENMD conformers shown on the angle space (LID-Core vs. NMP-Core angles) of adenylate kinase (AK), together with homologous experimental structures (black circles). 
          Independent 5-generation runs starting from open (4ake) and closed (1ake) states of AK highlight the major minima and the populated transition states. 
        </p>
      </div>

      <div class="prody-slide" id="slide-44">
        <div class="slide-image"><img src="../_static/figures/clustenmd_fig_2.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Conformational surface of HIV-1 reverse transcriptase plotted along the first two principal components (PCs) obtained from experimental structures (black circles), onto which the ClustENMD conformers 
          (red circles, population levels in cyan) are projected. 
        </p>
      </div>

      <div class="prody-slide" id="slide-45">
        <div class="slide-image"><img src="../_static/figures/clustenmd_fig_3.gif"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          This movie shows the population distribution for successive generations (gen-1 to gen-10) of conformers sampled starting from the open (blue levels; initial structure/black circle: 1tw7) and the closed 
          (red levels; initial structure/black diamond: 1bve) states of HIV-1 protease. As the distributions merge, they also cover the homologous experimental structures (gray circles). 
        </p>
      </div>

      <div class="prody-slide" id="slide-46">
        <div class="slide-image"><img src="../_static/gallery/insty1.jpg"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Detecting non-covalent interactions in  USP5 zinc-finger ubiquitin binding domain (PDB:7ms7). InSty detected five types of interactions 
          (HBs - hydrogen bonds, SBs - salt bridges, RIB - repulsive ionic bonding, PiStack - pi-stacking, PiCat - pi-cation). Such visualization is available 
          after loading the TCL file(s) (generated by InSty) into the VMD program. The structure is color-coded by the number of interactions (blue-white-red, 
          where red denotes the biggest number of interactions and blue the fewest).
        </p>
      </div>

      <div class="prody-slide" id="slide-47">
        <div class="slide-image"><img src="../_static/gallery/insty2.jpg"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Ensemble analysis of interactions. The upper panel displays the time evolution of interactions for detected types of interactions. The lower panel 
          shows the interaction pairs of the selected type of interaction. In this case, for hydrogen bonds. The color of the line corresponds to the 
          duration/frequency of interactions (in trajectory or PDB Ensemble) and the length to the distance between pairs or residues. 
          Histograms of distance and angle can be displayed for selected pairs of residues.
        </p>
      </div>

      <div class="prody-slide" id="slide-48">
        <div class="slide-image"><img src="../_static/gallery/insty3.jpg"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Predicting the number and types of interactions for each residue in the protein structure.
        </p>
      </div>

      <div class="prody-slide" id="slide-49">
        <div class="slide-image"><img src="../_static/gallery/watfinder1.jpg"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Detecting pairs of residues linked by water bridges (in red) in LMW-PTP protein  (PDB: 5KQM). Involved water molecules are displayed. In the lower panel, an example of a pair of residues, D92 and R18, 
          frequently interacting via water molecules. The number of interacting molecules is displayed on the histogram.</p>
      </div>

      <div class="prody-slide" id="slide-50">
        <div class="slide-image"><img src="../_static/gallery/watfinder2.jpg"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Determining prevalent water-attracted regions (water clusters) in the PE-binding protein 1 (PDB: 1BEH) structure based on PDB Ensemble.</p>
      </div>

      <div class="prody-slide" id="slide-51">
        <div class="slide-image"><img src="../_static/gallery/watfinder3.jpg"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
         Frequency of interactions with water molecules for a LMW-PTP protein (PDB: 5KQM) trajectory. Imshow maps provide additional information 
          about water-bridging residues (distance standard deviation, percentage of interaction, and average distance between pairs of residues).</p> 
      </div>

      <div class="prody-slide" id="slide-52">
        <div class="slide-image"><img src="../_static/gallery/watfinder4.jpg"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          The identification of the main water influx and clusters of water in the vesicular monoamine transporter VMAT2.</p>
      </div>

      <div class="prody-slide" id="slide-53">
        <div class="slide-image"><img src="../_static/gallery/watfinder5.jpg"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          The identification of water influx in the vesicular monoamine transporter VMAT2, with less restricted parameters,
          predicts two possible channels.</p>
      </div>

      <div class="prody-slide" id="slide-54">
        <div class="slide-image"><img src="../_static/figures/scipion_figure2_spike_overview.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Example workflow using ProDy within Scipion showing an ensemble analysis with 3 spike structures from the 
          Scipion-EM-ProDy paper. The panel (a) lists possible protocols and selecting one opens a form as shown in (b).
          Executing the protocols creates boxes in the workflow with outputs of one being inputs for the next as in (c).
          Some of them have associated viewers, such as the normal mode viewer using NMWiz (d) and the projection viewer (e).
        </p>
      </div>

      <div class="prody-slide" id="slide-55">
        <div class="slide-image"><img src="../_static/figures/scipion_figure5_gnm.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          An example GNM analysis in Scipion-EM-ProDy is shown, including visualisation of 
          the main results (a-c), a pipeline including dynamical domain decomposition and 
          comparison of all-atom and CA-only GNM (d), and visualisation of their results (e-g).
        </p>
      </div>

      <div class="prody-slide" id="slide-56">
        <div class="slide-image"><img src="../_static/figures/movie1_mGluR_mode1_anmd.gif"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Example output from ANMD analysis of the metabotropic glutamate receptor mGluR1 N-terminal
          domain showing the generated conformers along mode 1, closing the venus fly trap clamshell.
        </p>
      </div>

      <div class="prody-slide" id="slide-57">
        <div class="slide-image"><img src="../_static/figures/movie2_mGluR_mode2_anmd.gif"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          Example output from ANMD analysis of the metabotropic glutamate receptor mGluR1 N-terminal
          domain showing the generated conformers along mode 1, twisting the venus fly trap clamshell 
          as in related GABA-B receptors.
        </p>
      </div>

      <div class="prody-slide" id="slide-58">
        <div class="slide-image"><img src="../_static/figures/anmd_fig_allatom.png"/></div>
        <p></p><p style="font-size: 95%; font-style: italic;">
          The motion of the Omicron RBD (green) – V<sub>H</sub> F6 (red) complex, generated using ANMD 
          and sampled along ANM Mode 1 (two shades of cyan, softest mode of motion), enabled quantification of 
          intrinsic dynamics leading to variability at the interaction interface. The structures are superimposed on 
          the RBD to highlight variations at the V<sub>H</sub> F6 interface, facilitating an efficient evaluation of 
          changes in interaction affinity between the RBD and V<sub>H</sub> F6 across Omicron and Wild Type variants.
        </p>
      </div>

  <div class="container" id="about">

<!-- Logo & Nav
================================================== -->
    <div class="row">
      <div class="span12">
        <img src="{{ pathto('_static/logo.png', 1) }}" alt="ProDy logo" id="logo" />
      </div>
    </div>
      <div class="subnav" id="homepagenav">
        <ul class="nav nav-pills">
          <li title="ProDy"><a href="{{ pathto('', 1) }}">ProDy</a></li>
          <li title="Evol"><a href="{{ pathto('evol', 1) }}">Evol</a></li>
          <li title="NMWiz" ><a href="{{ pathto('nmwiz', 1) }}">NMWiz</a></li>
          <li title="SignDy" ><a href="{{ pathto('signdy', 1) }}">SignDy</a></li>
          <li title="membrANM" ><a href="{{ pathto('memanm', 1) }}">membrANM</a></li>
          <li title="stiffMech" ><a href="{{ pathto('mechstiff', 1) }}">MechStiff</a></li>
          <li title="PRS" ><a href="{{ pathto('prs', 1) }}">PRS</a></li>
          <li title="DruGUI" ><a href="{{ pathto('drugui', 1) }}">DruGUI</a></li>
          <li title="Pharmmaker" ><a href="{{ pathto('pharmmaker', 1) }}">Pharmmaker</a></li>
          <li title="coMD" ><a href="{{ pathto('comd', 1) }}">coMD</a></li>
          <li title="ESSA" ><a href="{{ pathto('essa', 1) }}">ESSA</a></li>  
          <li title="CryoDy" ><a href="{{ pathto('cryody', 1) }}">CryoDy</a></li>
          <li title="ClustENMD" ><a href="{{ pathto('clustenmd', 1) }}">ClustENMD</a></li>
          <li title="InSty" ><a href="{{ pathto('insty', 1) }}">InSty</a></li>
          <li title="WatFinder" ><a href="{{ pathto('watfinder', 1) }}">WatFinder</a></li>
          <li title="Scipion" ><a href="{{ pathto('scipion', 1) }}">Scipion</a></li>
          <li title="ANMD" ><a href="{{ pathto('anmd', 1) }}">ANMD</a></li>
          <li title="Downloads"><a href="{{ pathto('downloads', 1) }}">Downloads</a></li>
          <li title="Tutorials"><a href="{{ pathto('tutorials', 1) }}">Tutorials</a></li>
          <li title="Workshops"><a href="{{ pathto('workshop', 1) }}">Workshops</a></li>
          <li title="Statistics"><a href="{{ pathto('statistics', 1) }}">Statistics</a></li>
          <li class="pull-right">
            <form class="navbar-search form-search pull-right" action="{{ pathto('search') }}" method="get" style="margin-right: 15px">
              <div class="input-append">
                <input type="text" name="q" class="search-query input-large" placeholder="Search Manual and Tutorials">
                <button type="submit" class="btn btn-primary">Go</button>
              </div>
              <input type="hidden" name="check_keywords" value="yes" />
              <input type="hidden" name="area" value="default" />
            </form>
          </li>
        </ul>
      </div>

<!-- Body
================================================== -->
{% block body %}
{% endblock %}
<!-- Footer
================================================== -->

          <div class="row">
            <div class="span3">
              <h3>People</h3>
              <p><i>ProDy</i> is developed in <a href="http://bahargroup.org/Faculty/bahar/" target="_blank">Bahar Lab</a>
	      at the <a href="https://laufercenter.stonybrook.edu" target="_blank">Laufer Center, Stony Brook University</a>.
	       It was originally launched and developed at the <a href="http://pitt.edu" target="_blank">University of Pittsburgh</a>.
                Click <a href="{{ pathto('manual/about/people') }}">here</a> to see a list of people who contributed to its development.</p>
            </div>
            <div class="span3">
              <h3>Community</h3>
              <p><i>ProDy</i> makes use of great open source software including
                <a href="http://www.numpy.org/" target="_blank">NumPy</a>,
                <a href="https://pyparsing-docs.readthedocs.io/en/latest/" target="_blank">Pyparsing</a>,
                <a href="http://biopython.org/" target="_blank">Biopython</a>,
                <a href="http://scipy.org/" target="_blank">SciPy</a>,
                and
                <a href="http://matplotlib.org/" target="_blank">Matplotlib</a>.
                Click <a href="{{ pathto('manual/about/credits') }}">here</a> for details.</p>
            </div>
            <div class="span3">
              <h3>Source Code</h3>
              <p><i>ProDy</i> is open <a href="https://github.com/prody/ProDy">source</a>
                and you can contribute to its development in many ways.
                See this <a href="{{ pathto('manual/devel/develop') }}">guide</a> for getting started.
            </div>
            <div class="span3">
              <h3>Problems?</h3>
              <p>Let us know any problems you might have by opening an issue at the
                <a href="https://github.com/prody/ProDy/issues">tracker</a>
                so that we can make <i>ProDy</i> better.
              </p>
            </div>
          </div>
  <footer class="footer">
    <div class="row">
      <div class="span4 text-left">
        <!--
        <a href="https://pypi.python.org/pypi/ProDy"><img src="https://pypip.in/v/ProDy/badge.png"></a>
        <a href="https://crate.io/packages/ProDy/"><img src="https://pypip.in/d/ProDy/badge.png"></a>
      -->
      </div>
      <div class="span8">
        <p><small>
        &copy; Copyright 2010-2014, University of Pittsburgh. Last updated on {{ last_updated }}.
        </small></p>
      </div>
    </div>
  </footer>

</div>

  </body>
</html>
